In [8]:
import pandas as pd

train_df = pd.read_csv("./ssec/val.csv", error_bad_lines=False, header=None, delimiter="\t")
train_df

/Users/ItishaYadav1/opt/anaconda3/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3444: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version.


  exec(code_obj, self.user_global_ns, self.user_ns)


0             1        2     3    4        5         6      7  \
0    Anger  Anticipation  Disgust   ---  Joy      ---       ---    ---   
1      ---  Anticipation      ---   ---  Joy      ---  Surprise    ---   
2    Anger  Anticipation  Disgust  Fear  ---  Sadness       ---    ---   
3    Anger           ---      ---   ---  Joy      ---       ---  Trust   
4    Anger  Anticipation  Disgust   ---  ---  Sadness       ---    ---   
..     ...           ...      ...   ...  ...      ...       ...    ...   
287  Anger           ---  Disgust   ---  ---  Sadness       ---    ---   
288    ---           ---      ---   ---  ---  Sadness       ---  Trust   
289  Anger  Anticipation  Disgust   ---  ---  Sadness       ---    ---   
290  Anger           ---  Disgust   ---  ---  Sadness       ---    ---   
291  Anger           ---  Disgust   ---  ---      ---  Surprise    ---   

                                                     8  
0    on a side note, just because you think smtg is...  
1    Woah the hashtag puts the rainbow heart itself...  
2    I refuse 2comply w/Bad laws. Just Because #SCO...  
3    Just owned a woman who works for planned paren...  
4    If it's getting old maybe you should stop call...  
..                                                 ...  
287  There's a law protecting unborn eagles, but no...  
288  I am 1 in 3... I have had an abortion #Abortio...  
289  How dare you say my sexual preference is a cho...  
290  Equal rights for those 'born that way', no rig...  
291  #POTUS seals his legacy w/ 1/2 doz wins. The #...  

[292 rows x 9 columns]

# Data Details # Multi-label-classification
# Emotions tags:
Anger, Anticipation, Disgust, Fear, Joy, Sadness, Surprise, Trust
train.csv = 2622
test.csv = 1956
val.csv = 292

In [72]:
import pandas as pd
isear_train_df = pd.read_csv("./isear/isear-train.csv", error_bad_lines=False, header=None)
isear_train_df

/Users/ItishaYadav1/opt/anaconda3/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3444: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version.


  exec(code_obj, self.user_global_ns, self.user_ns)
b'Skipping line 119: expected 2 fields, saw 7\nSkipping line 1213: expected 2 fields, saw 4\nSkipping line 2323: expected 2 fields, saw 3\nSkipping line 2803: expected 2 fields, saw 3\nSkipping line 3630: expected 2 fields, saw 4\nSkipping line 4635: expected 2 fields, saw 5\nSkipping line 4797: expected 2 fields, saw 4\n'


0                                                  1
0         joy  When I understood that I was admitted to the U...
1        fear  I broke a window of a neighbouring house and I...
2         joy                         Got a big fish in fishing.
3        fear  Whenever I am alone in a dark room, walk alone...
4       shame  I bought a possible answer to a homework probl...
...       ...                                                ...
5328  sadness  After a disagreement between my parents, when ...
5329    guilt  One day I shouted at my brother who didn't do ...
5330     fear  Before one of my final exams of the third year...
5331  disgust  A drunk man bumped into me and wanted to grip ...
5332    shame  Had a very good friend. We grew apart I found ...

[5333 rows x 2 columns]

In [64]:
mapping = {"fear": 1, "anger": 2, "guilt": 3, "joy": 4, "shame": 5, "disgust": 6, "sadness": 7}

In [52]:
isear_train_df[0].value_counts() #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

joy                                                                                                                                777
guilt                                                                                                                              766
sadness                                                                                                                            760
anger                                                                                                                              758
shame                                                                                                                              757
disgust                                                                                                                            757
fear                                                                                                                               751
Note: for GUILT go to record n¦231"                    

# Data Details # Multi-class-classification
# Emotions tags:
joy, guilt, sadness, anger, shame, disgust, fear
train.csv = 5333
test.csv = 1146
val.csv = 1150

In [92]:
# Recall
import numpy as np


class evaluation:
    def __init__(self, y_actual, y_pred):
        self.y_actual = np.array(y_actual)
        print(self.y_actual)
        self.y_pred = np.array(y_pred)
        print(self.y_pred)

    def confusion_matrix(self, actual, pred):
        tp = fp = tn = fn = 0
        for i, j in zip(actual, pred):
            if i == 1:
                # positive
                if i == j:
                    tp += 1
                else:
                    fp += 1
            else:
                # negative
                if i == j:
                    tn += 1
                else:
                    fn += 1
        cf = pd.DataFrame([[tp, fp], [fn, tn]], columns=["actual_pos", "actual_neg"], index=["pred_pos", "pred_neg"])
        return cf, tp, fp, tn, fn

    def recall(self, actual, pred):
        cf, tp, fp, tn, fn = self.confusion_matrix(actual, pred)
        return tp / (tp+fn)

    def precision(self, actual, pred):
        cf, tp, fp, tn, fn = self.confusion_matrix(actual, pred)
        return tp / (tp+fp)

    def f1(self, actual, pred):
        # harmonic mean
        pr = self.precision(actual, pred)
        re = self.recall(actual, pred)
        f1 = 2 * ((pr * re) / (pr + re))
        return f1

    def main(self):
        mapping = {1: "fear", 2: "anger", 3: "guilt", 4: "joy", 5: "shame", 6: "disgust", 7: "sadness"}
        for cls in [1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ,7.0]:
            print("Class : ", mapping[cls])
            mod_y_actual = []
            for i in self.y_actual:
                if i == cls:
                    mod_y_actual.append(1)
                else:
                    mod_y_actual.append(0)
            mod_y_pred = []
            for i in self.y_pred:
                if i == cls:
                    mod_y_pred.append(1)
                else:
                    mod_y_pred.append(0)
            #print(mod_y_pred)
            print()
            print()
            print("Confusion Matrix : \n", self.confusion_matrix(mod_y_actual, mod_y_pred)[0])
            print("*******************************************************\n")
            print("Precision : \n", self.precision(mod_y_actual, mod_y_pred))
            print("*******************************************************\n")
            print("Recall : \n", self.recall(mod_y_actual, mod_y_pred))
            print("*******************************************************\n")
            print("F1 Score : \n", self.f1(mod_y_actual, mod_y_pred))
            print()
            print()








In [90]:
import pandas as pd
isear_train_df = pd.read_csv("./isear/isear-train.csv", error_bad_lines=False, header=None)
isear_train_df

mapped_emotions = isear_train_df[0].map(mapping)
isear_train_df[0] = mapped_emotions

isear_train_df

/Users/ItishaYadav1/opt/anaconda3/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3444: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version.


  exec(code_obj, self.user_global_ns, self.user_ns)
b'Skipping line 119: expected 2 fields, saw 7\nSkipping line 1213: expected 2 fields, saw 4\nSkipping line 2323: expected 2 fields, saw 3\nSkipping line 2803: expected 2 fields, saw 3\nSkipping line 3630: expected 2 fields, saw 4\nSkipping line 4635: expected 2 fields, saw 5\nSkipping line 4797: expected 2 fields, saw 4\n'


0                                                  1
0     4.0  When I understood that I was admitted to the U...
1     1.0  I broke a window of a neighbouring house and I...
2     4.0                         Got a big fish in fishing.
3     1.0  Whenever I am alone in a dark room, walk alone...
4     5.0  I bought a possible answer to a homework probl...
...   ...                                                ...
5328  7.0  After a disagreement between my parents, when ...
5329  3.0  One day I shouted at my brother who didn't do ...
5330  1.0  Before one of my final exams of the third year...
5331  6.0  A drunk man bumped into me and wanted to grip ...
5332  5.0  Had a very good friend. We grew apart I found ...

[5333 rows x 2 columns]

In [91]:
print(isear_train_df.shape)
obj = evaluation(isear_train_df[0], isear_train_df[0])
obj.main()

(5333, 2)
[4. 1. 4. ... 1. 6. 5.]
[4. 1. 4. ... 1. 6. 5.]
Class :  fear


Confusion Matrix : 
           actual_pos  actual_neg
pred_pos         751           0
pred_neg           0        4582
*******************************************************

Precision : 
 1.0
*******************************************************

Recall : 
 1.0
*******************************************************

F1 Score : 
 1.0


Class :  anger


Confusion Matrix : 
           actual_pos  actual_neg
pred_pos         758           0
pred_neg           0        4575
*******************************************************

Precision : 
 1.0
*******************************************************

Recall : 
 1.0
*******************************************************

F1 Score : 
 1.0


Class :  guilt


Confusion Matrix : 
           actual_pos  actual_neg
pred_pos         766           0
pred_neg           0        4567
*******************************************************

Precision : 
 1.0
***********

In [57]:
isear_train_df

0                                                  1
0     4.0  When I understood that I was admitted to the U...
1     1.0  I broke a window of a neighbouring house and I...
2     4.0                         Got a big fish in fishing.
3     1.0  Whenever I am alone in a dark room, walk alone...
4     5.0  I bought a possible answer to a homework probl...
...   ...                                                ...
5328  7.0  After a disagreement between my parents, when ...
5329  3.0  One day I shouted at my brother who didn't do ...
5330  1.0  Before one of my final exams of the third year...
5331  6.0  A drunk man bumped into me and wanted to grip ...
5332  5.0  Had a very good friend. We grew apart I found ...

[5333 rows x 2 columns]